homepage/migrations/0003_course_test_data.py

```diff
@@ -0,0 +1,37 @@
+from django.db import migrations, transaction
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('homepage', '0001_initial'),
+    ]
+
+    def generate_data(apps, schema_editor):
+        from homepage.models import Course
+        from homepage.models import Prerequisites
+
+        test_data_course = [
+            (10231, 'UnFogging the Future', 1, 4, 'Currently Unavailable', 3.25, 1.5, 12, 12),
+            (10221, 'Grammatica in Arithmancy', 0, 3, 'Currently Unavailable', 4.25, 4.5, 10, 9),
+            (10111, 'Resonance in Runes and Signs', 1, 3, 'Currently Unavailable', 2.25, 2.0, 13, 13),
+            (12357, 'Numerology', 0, 5, 'Currently Unavailable', 3.5, 2.5, 7, 3),
+        ]
+
+        courses = [Course(*tdc) for tdc in test_data_course]
+        test_data_prerequisites = [
+            (courses[3], courses[0], 1),
+            (courses[2], courses[1], 0),
+            ]
+
+        with transaction.atomic():
+            for course in courses:
+                course.save()
+
+            for preq in test_data_prerequisites:
+                Prerequisites(course_id=preq[0], req_course_id=preq[1], req_code=preq[2]).save()
```

Comment:
its not too much parameters so just if you want you can do it like
```suggestion
                Prerequisites(*preq).save()
```

Reply:
I tried it when I was fixing this file, but it didn't work then and doesn't work now either, and I'm not sure why, but for some reason this interrupts the Foreign Key part of the definition - I get the error:
`TypeError: Field 'id' expected a number but got <Course: Course object (12357)>.`
If I change the test_data_prerequisites to contain courses[i].course_id it gives another error that indicates that this change prevents correct data placement: it says I have accidentally given a primary key to the instance of Prerequisites.
As such, I'd rather leave it as it is, because it works, and has only 3 parameters.

Reply:
I'm ok with this code too :)
But...
it looks like the preq[0] is:
            (courses[3], courses[0], 1)
so preq[0] is courses[3] which is Course object that initialized with the following parameters in line # 21:
            (12357, 'Numerology', 0, 5, 'Currently Unavailable', 3.5, 2.5, 7, 3),
so if you want only the course_id you should use it only something like:
test_data_prerequisites = [
            (courses[3].course_id, courses[0].course_id, 1),
            (courses[2].course_id, courses[1].course_id, 0),
            ]

Reply:
Thank you for replying on the weekend, it's really apriciated! 

I tried exactly that but:
- with the way the prerequisites are saved in the current code I get the error that it expects an object for the foreign key and I'm sending it an integer
- with the Prerequisites(*args) way it somehow overrides the automated PK and puts the last thing in the tuple (0 for example) as the req_course_id which is wrong

I'm really not sure why at all, couldn't find a good enough query for google.

Reply:
I figured it out, by using this code:
`        test_data_prerequisites = [`
`            (1, courses[3].course_id, courses[0].course_id, 1),`
`            (2, courses[2].course_id, courses[1].course_id, 0),
            ]`
then this works:
`            for preq in test_data_prerequisites:`
                `Prerequisites(*preq).save()`

It means I overrode the auto-PK this is supposed to produce, so I don't know if this is best practice for test data?
If it's okay, I'll commit the new file.